core/src/services/s3/writer.rs

```diff
@@ -158,7 +158,16 @@ impl oio::MultipartWrite for S3Writer {
         let status = resp.status();
 
         match status {
-            StatusCode::OK => Ok(()),
+            StatusCode::OK => {
+                let (_, mut body) = resp.into_parts();
+                let bs = body.copy_to_bytes(body.remaining());
+                let result: CompleteMultipartUploadResult =
```

Comment:
Maybe we can use `parse_error` while the returning response contains `<Error>`? Using `Error::new(ErrorKind::Unexpected, &result.message)` here is incorrect since we have our `temporary` and `context` to set.

Reply:
I refactor the logic in 1bc1e55, which first decode an `S3Error` and check its content, on the error branch it then convert the `S3Error` into `Error`.

Reply:
I once tried to reuse `from_s3_error()` in `parse_error()` but it needs to pass many local mutable variables as parameters. So I ended up making a new one (though I suppose no other interface will need this special check on 200...

Reply:
Looks good to me now!